src/Functions/JSONPaths.cpp

```diff
@@ -0,0 +1,510 @@
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        const ColumnWithTypeAndName & elem = arguments[0];
+        const auto * column_object = typeid_cast<const ColumnObject *>(elem.column.get());
+        if (!column_object)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unexpected column type in function {}. Expected Object column, got {}", getName(), elem.column->getName());
+
+        const auto & type_object = assert_cast<const DataTypeObject &>(*elem.type);
+        if constexpr (Impl::with_types)
+            return executeWithTypes(*column_object, type_object);
+        return executeWithoutTypes(*column_object);
+    }
+
+private:
+    ColumnPtr executeWithoutTypes(const ColumnObject & column_object) const
+    {
+        if constexpr (Impl::paths_mode == PathsMode::SHARED_DATA_PATHS)
+        {
+            /// No need to do anything, we already have a column with all sorted paths in shared data.
+            const auto & shared_data_array = column_object.getSharedDataNestedColumn();
+            const auto & shared_data_paths = assert_cast<const ColumnTuple &>(shared_data_array.getData()).getColumnPtr(0);
+            return ColumnArray::create(shared_data_paths, shared_data_array.getOffsetsPtr());
+        }
+
+        auto res = ColumnArray::create(ColumnString::create());
+        auto & offsets = res->getOffsets();
+        ColumnString & data = assert_cast<ColumnString &>(res->getData());
+
+        if constexpr (Impl::paths_mode == PathsMode::DYNAMIC_PATHS)
+        {
+            /// Collect all dynamic paths.
+            const auto & dynamic_path_columns = column_object.getDynamicPaths();
+            std::vector<String> dynamic_paths;
+            dynamic_paths.reserve(dynamic_path_columns.size());
+            for (const auto & [path, _] : dynamic_path_columns)
+                dynamic_paths.push_back(path);
+            /// We want the resulting arrays of paths to be sorted for consistency.
+            std::sort(dynamic_paths.begin(), dynamic_paths.end());
+
+            size_t size = column_object.size();
+            for (size_t i = 0; i != size; ++i)
+            {
+                for (const auto & path : dynamic_paths)
+                {
+                    /// Don't include path if it contains NULL, because we consider
+                    /// it to be equivalent to the absence of this path in this row.
+                    if (!dynamic_path_columns.at(path)->isNullAt(i))
+                        data.insertData(path.data(), path.size());
+                }
+                offsets.push_back(data.size());
+            }
+            return res;
+        }
+
+        /// Collect all paths: typed, dynamic and paths from shared data.
+        std::vector<String> sorted_dynamic_and_typed_paths;
```

Comment:
```suggestion
        std::vector<std::string_view> sorted_dynamic_and_typed_paths;
```

Reply:
We use values of this vector as keys in `std::unordered_map<String>`, so unfortunately we cannot use `std::string_view` here, or we will have to construct `String` from `std::string_view` every time we want to do a lookup in `unordered_map`

Reply:
There is something called `Heterogeneous lookup`.
Reference: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0919r3.html
Example of implementation: https://github.com/ClickHouse/ClickHouse/blob/master/src/Coordination/KeeperStorage.h#L612

Reply:
Wow, good to know, thanks! Will use it in all similar places